File: MusicXmlDb.Server/MusicXmlDb.Server/ScoreDocuments/ScoreDocument.cs
using MusicXmlDb.Server.Users;

namespace MusicXmlDb.Server.ScoreDocuments;

public class ScoreDocument
{
    public Guid Id { get; set; }

    public string UserId { get; set; } = "";

    public ICollection<ScoreDocumentHistory> History { get; set; } = [];

    public string Name { get; set; } = "";
    public int Views { get; set; }
    public DateTime Created { get; set; } = DateTime.MinValue;
    public DateTime Modified { get; set; } = DateTime.MinValue;
    public bool IsPublic { get; set; }
}

File: MusicXmlDb.Client/src/models/ScoreDocument.ts
import type { ScoreDocumentHistory } from "./ScoreDocumentHistory";

export interface ScoreDocument {
    id: string;
    userId: string;
    history: ScoreDocumentHistory[];
    name: string;
    views: number;
    created: Date;
    modified: Date;
    isPublic: boolean;
}


